src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs

```diff
@@ -267,16 +269,35 @@ public override async ValueTask<FormulaValue> Visit(CallNode node, EvalVisitorCo
             {                
                 return await asyncConnectorTexlFunction.InvokeAsync(args, _services, _cancellationToken).ConfigureAwait(false);
             }
-            else if (func is UserDefinedTexlFunction udtf)
-            {
-                // $$$ Should add _runtimeConfig
-                result = await udtf.InvokeAsync(args, _cancellationToken, context.StackDepthCounter.Increment()).ConfigureAwait(false);
-            }
             else if (func is CustomTexlFunction customTexlFunc)
             {
                 // If custom function throws an exception, don't catch it - let it propagate up to the host.
                 result = await customTexlFunc.InvokeAsync(FunctionServices, args, _cancellationToken).ConfigureAwait(false);
             }
+            else if (func is UserDefinedFunction userDefinedFunc)
+            {
+                var recordValue = userDefinedFunc.GetRuntimeRecordValue(args, _cancellationToken);
+
+                var symbolsTableRecord = ReadOnlySymbolTable.NewFromRecord(recordValue.Type);
+                var symbolValuesRecord = SymbolValues.NewFromRecord(symbolsTableRecord, recordValue);
+
+                var symbolsTable = ReadOnlySymbolTable.Compose(_symbolValues.SymbolTable, symbolsTableRecord);
+                var symbolValues = ReadOnlySymbolValues.Compose(_symbolValues, symbolValuesRecord);
+
+                userDefinedFunc.SetFormulaValues(args);
+                userDefinedFunc.BindBody(symbolsTable, new Glue2DocumentBinderGlue(), bindingConfig: BindingConfig.Default);
+
+                (var irnode, var ruleScopeSymbol) = userDefinedFunc.GetIRTranslator();
+
+                var expr = new ParsedExpression(irnode, ruleScopeSymbol, context.StackDepthCounter)
+                {
+                    _allSymbols = symbolsTable
+                };
+
+                RuntimeConfig runtimeConfig = new RuntimeConfig(symbolValues, CultureInfo);
+
+                result = await expr.EvalAsync(_cancellationToken, runtimeConfig).ConfigureAwait(false);

```

Comment:
>expr [](http://example.com/codeflow?start=31&length=4)

we probably also want to sue the same EvalVisitor on the udf.  (similar to what we do when running a lambda)
It has too much state (service provider, current culture, etc). 


For example, a test case is:

`Temp() : string => Language();`

Then this should return true:
`Language() = Temp()`